lib/spider/scripts/in_context_learning.py

```diff
@@ -57,15 +65,35 @@ def main(cfg) -> None:
         }
         for data_point in dev_json
     ]
+    if cfg.prompts.template.count("{}") > 2:
+        train_json = json.load(open(DATASET_PATH / "train_spider.json"))
+        train_json = [
+            {
+                "db_id": data_point["db_id"],
+                "question": data_point["question"],
+                "query": data_point["query"],
+            }
+            for data_point in train_json
+        ]
+        train_questions = [data_point["question"] for data_point in train_json]
+        top_k = int((cfg.prompts.template.count("{}") - 2) / 2)
+        encoder_model = SentenceTransformer("all-mpnet-base-v2")
+        encoder_model.eval()
+        with torch.no_grad():
+            train_embeddings = encoder_model.encode(train_questions, convert_to_tensor=True)
 
     # Model loading
     bnb_config: BitsAndBytesConfig = hydra.utils.instantiate(cfg.bnb_config)
+    # if you use flash attention, you need to run:
+    # pip install flash-attn --no-build-isolation in the poerty shell
     model = AutoModelForCausalLM.from_pretrained(
         cfg.pretrained_model_name_or_path,
         device_map="auto",
         quantization_config=bnb_config,
         torch_dtype=torch.bfloat16,
+        attn_implementation="flash_attention_2",
```

Comment:
Can the `attn_implementation` parameter be passed directly in one of the hydra configs?